apollo-federation/src/query_graph/path_tree.rs

```diff
@@ -292,7 +292,10 @@ where
             match for_edge.by_unique_trigger.entry(trigger) {
                 Entry::Occupied(entry) => {
                     let existing = entry.into_mut();
-                    existing.trigger = trigger;
+                    if *trigger < *existing.trigger {
+                        // overwrite if the new trigger is preferable (= less)
```

Comment:
I think it would be better to just have a nicely named method here instead of using an overloaded operator that requires a comment at every usage site ^^